app/screens/modal/PoSProvingOptsUpdateWarningModal.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PoSProfiler from '../../components/node/PoSProfiler';
+import { Modal } from '../../components/common';
+import { getWarningByType } from '../../redux/ui/selectors';
+import { WarningObject, WarningType } from '../../../shared/warning';
+import { omitWarning } from '../../redux/ui/actions';
+import { smColors } from '../../vars';
+import useUpdatePostProvingOpts from '../../hooks/useUpdatePostProvingOpts';
+import ReactPortal from './ReactPortal';
+
+const PoSProvingOptsUpdateWarningModal = () => {
+  const dispatch = useDispatch();
+  const provingOptsWarning = useSelector(
+    getWarningByType(WarningType.UpdateSmeshingProvingOpts)
+  );
+  const {
+    updateConfigHandler,
+    numUnits,
+    dataDir,
+    singleCommitmentSize,
+    loading,
+    nonces,
+    threads,
+  } = useUpdatePostProvingOpts(() => {
+    dispatch(
+      omitWarning(
+        provingOptsWarning as WarningObject<WarningType.UpdateSmeshingProvingOpts>
+      )
+    );
+  });
```

Comment:
Please, trust TypeScript more ;)
```suggestion
  } = useUpdatePostProvingOpts(() => {
    provingOptsWarning && dispatch(omitWarning(provingOptsWarning));
  });
```